### ISSUE TITLE
Syntax-check does not print file name if it fails to parse

### ISSUE THREAD

**Issue by hughsaunders** [ansible/ansible#9703]
##### Summary:

If you specify a list of playbooks to ansible-playbook --syntax-check and one fails on parsing, ansible-playbook will halt but not tell you which file caused the failure. This is annoying if you are checking a large set of playbooks. 
##### Steps To Reproduce:

```
$ tail *.yml
==> a.yml <==

---
- hosts: localhost
  tasks:
    - ping:


==> b.yml <==

---
hosts: localhost
ping:


$ ansible-playbook -i ../local_only_inventory --syntax-check *.yml

playbook: a.yml


ERROR: parse error: playbooks must be formatted as a YAML list, got <type 'dict'>
```

The above example show two playbooks, the syntax of a.yml is fine, b.yml is invalid. Syntax check does not print that the error is with b.yml. 
##### Expected Results:

```
$ tail *.yml
==> a.yml <==

---
- hosts: localhost
  tasks:
    - ping:


==> b.yml <==

---
hosts: localhost
ping:


$ ansible-playbook -i ../local_only_inventory --syntax-check *.yml

playbook: a.yml

playbook: b.yml
ERROR: parse error: playbooks must be formatted as a YAML list, got <type 'dict'>
```
##### Actual Results:

See steps to reproduce

**Comment by srri-zz** [comment]
Fixed in #9741, new output is:

```
playbook: a.yml

ERROR: b.yml
parse error: playbooks must be formatted as a YAML list, got <type 'dict'>
```

**Comment by resmo** [comment]
could not reproduce with 2.1.1, foo.yml is correct, bla.yml not

```
 $ ansible-playbook --syntax-check foo.yml bla.yml 
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available

ERROR! 'debug' is not a valid attribute for a Play

The error appears to have been in '/home/resmo/tmp/xyz/bla.yml': line 2, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
- hosts: bla
  ^ here
```

closing
